src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -3990,5 +4028,113 @@ public void AssertNotDisposed()
             if (_disposeOne.Disposed)
                 ThrowObjectDisposed();
         }
+
+        internal virtual IEnumerable<IIndexCollection> GetCollectionsForIndexing()
```

Comment:
Try to get rid of this and change `Collections` hashset to use `IIndexCollection`